Fix some typos (a-c)

janekdb opened the pull request.

Ichoran commented:
LGTM

janekdb commented:
/rebuild

mpociecha commented:
The build failed. I think that I have corrections in my local branch (I used text search to check whether there are no more typos like these corrected by you, and fixed some things by the way). I'm going to create PR against your branch in your fork, when `ant test` ran locally will finish.

mpociecha committed:
Fix another several typos

I just used text search to check whether there are no more typos like
these corrected by janekdb, and by the way fixed also some other ones
which I saw.

mpociecha commented:
Ah, right - in scala/scala each commit should work separately. Therefore, I just changed `names-defaults-neg.check` directly in your commit (and I'm not creating this mentioned PR). If you want, you could just rewrite the history of your branch using my branch: https://github.com/mpociecha/scala/tree/2.11.x-scaladoc-2-ext There's also my additional commit with corrected several other typos. Thanks. :)

janekdb commented:
The corrected commit in now the tip of https://github.com/janekdb/scala/tree/2.11.x-scaladoc-3 

Should I abandon this PR and create a new PR from the branch above?

Or should I be looking to modify the commit in this PR?

If the second option is the best approach then how can this be done? My attempt to push up a new version of the commit failed,

```
$ git push origin HEAD:2.11.x-scaladoc-2
To git@github.com:janekdb/scala.git
 ! [rejected]        HEAD -> 2.11.x-scaladoc-2 (non-fast-forward)
error: failed to push some refs to 'git@github.com:janekdb/scala.git'
hint: Updates were rejected because the tip of your current branch is behind
```

If I then fetch and rebase I get two commits,

```
$ git fetch
$ git rebase origin/2.11.x-scaladoc-2
```

```
$ git log -3 --oneline
80f0e4d Fix 21 typos (a-c)
c658d8a Fix 19 typos (a-c)
6b09c66 Merge pull request #4548 from ScrapCodes/git_4522
```

If I then squash the first two and try to push the non-fast-forward warning is issued again.

```
$ git rebase -i HEAD~2

pick c658d8a Fix 19 typos (a-c)
squash 80f0e4d Fix 21 typos (a-c)

[detached HEAD 71d0727] Fix 21 typos (a-c)
 18 files changed, 21 insertions(+), 21 deletions(-)
Successfully rebased and updated refs/heads/2.11.x-scaladoc-2.

$ git log -3 --oneline
71d0727 Fix 21 typos (a-c)
6b09c66 Merge pull request #4548 from ScrapCodes/git_4522
39a5e1a SI-7747 More tests and logic according to our conclusions on #4522.

$ git push origin HEAD:2.11.x-scaladoc-2
To git@github.com:janekdb/scala.git
 ! [rejected]        HEAD -> 2.11.x-scaladoc-2 (non-fast-forward)
error: failed to push some refs to 'git@github.com:janekdb/scala.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

I may need some basic advice or a pointer here. Thanks!

mpociecha commented:
You could just:
- fetch my branch (I don't want to create another PR with these another several fixed typos while you can push it with your changes) ;)
- remove your old `2.11.x-scaladoc-2` branch (only locally)
- rename my branch to `2.11.x-scaladoc-2`
- push with -f to your private fork

Github will see changes and everything will be rebuilt by Jenkins.

mpociecha commented:
Right, I could change your commit message to 21 as it's not 19 anymore. :) It's corrected here https://github.com/mpociecha/scala/tree/2.11.x-scaladoc-2-ext2 (just in case - note that you're still an author)

Cheers,
Michał

janekdb commented:
Thanks for the tips. I pushed your `ext2` branch.

```
git remote add mpociecha git@github.com:mpociecha/scala.git
git fetch mpociecha
git push -f origin mpociecha/2.11.x-scaladoc-2-ext2:2.11.x-scaladoc-2
```

janekdb commented:
`integrate-ide` failed with

```
[scalacheck] [error] (*:update) sbt.ResolveException:
  unresolved dependency: org.apache.maven#maven-aether-provider;3.0.4: not found
```

The job lists 14 changes instead of the two in this PR.

SethTisue commented on test/files/jvm/bytecode-test-example/Test.scala:
```diff
@@ -17,7 +17,7 @@ object Test extends BytecodeTest {
     /** Is given instruction a null check?
      *  NOTE
-     *   This will detect direct null compparsion as in
+     *   This will detect direct null comparsion as in
```
should be `comparison`

SethTisue commented:
/rebuild

janekdb committed:
Fix some typos (a-c)

SethTisue commented:
thank you for this!
